syft_tensorflow/serde/serde.py

```diff
@@ -138,6 +140,171 @@ def _detail_tf_variable(worker, tensor_tuple) -> tf.Tensor:
+
+    layer_dict_ser = syft.serde.serde._simplify(layer_ser)
+
+    batch_input_shape_ser = syft.serde.serde._simplify(
+        layer_ser["config"]["batch_input_shape"]
+    )
+
+    chain = None
+    if hasattr(layer, "child"):
+        chain = syft.serde._simplify(layer.child)
+
+    return layer.id, layer_dict_ser, weights_ser, batch_input_shape_ser, chain
+
+
+def _detail_tf_keras_layers(worker, layer_tuple) -> tf.Tensor:
+    """
+    This function converts a serialized tf tensor into a local TF tensor
+    using tf.io.
+
+    Args:
+        tensor_tuple (bin): serialized obj of torch tensor. It's a tuple where
+            the first value is the ID, the second vlaue is the binary for the
+            TensorFlow object, the third value is the tensor_dtype_enum, and
+            the fourth value is the chain of tensor abstractions
+
+    Returns:
+        tf.Tensor: a deserialized TF tensor
+    """
+
+    layer_id, layer_bin, weights_bin, batch_input_shape_bin, chain = layer_tuple
+
+    layer_dict = syft.serde.serde._detail(worker, layer_bin)
+
+    layer = tf.keras.layers.deserialize(layer_dict)
+
+    weights = syft.serde.serde._detail(worker, weights_bin)
+
+    batch_input_shape = syft.serde.serde._detail(worker, batch_input_shape_bin)
+
+    layer.build(batch_input_shape)
+
+    layer.set_weights(weights)
+
+    initialize_object(
+        hook=syft.tensorflow.hook,
+        obj=layer,
+        owner=worker,
+        reinitialize=False,
+        id=layer_id,
+        init_args=[],
+        init_kwargs={},
+    )
+
+    if chain is not None:
+        chain = syft.serde._detail(worker, chain)
+        layer.child = chain
+        layer.is_wrapper = True
+
+    return layer
+
+
+def _simplify_keras_model(model: tf.keras.Model):
+    bio = io.BytesIO()
+
+    with h5py.File(bio) as file:
+        tf.keras.models.save_model(model, file, include_optimizer=True, overwrite=True)
+
+    model_ser = bio.getvalue()
+
+    return model_ser, model.id
+
+
+def _detail_keras_model(worker, model_tuple):
+    model_ser, model_id = model_tuple
+    bio = io.BytesIO(model_ser)
+    with h5py.File(bio) as file:
+        model = tf.keras.models.load_model(file)
+
+    initialize_object(
+        hook=syft.tensorflow.hook,
+        obj=model,
+        owner=worker,
+        reinitialize=False,
+        id=model_id,
+        init_args=[],
+        init_kwargs={},
+    )
+
+    return model
+
+
+def _simplify_keras_history_callback(
+        history_cb: tf.keras.callbacks.History
+    ) -> bin:
+    """
+    This function converts history callback into serialized dictionaries.
+
+    Args:
+      history_cb (tf.keras.callbacks.History): history callback
+
+    Returns:
+      tuple: serialized tuple of history callback. The first value is
+      the binary of the params dictionary. The second is the binary 
+      of the history dictionary.
+    """
+
+    params = history_cb.params
+    # serde can't serialize numpy values. These values need
```

Comment:
serde _can_ serialize numpy values, but right now it's in torch_serde.py -- I seem to remember that using pickle by default (which we might want to avoid), but either way I'd take another look at this and be sure this is the right approach

Reply:
but the right approach could just be to move that ndarray simplifier/detailer into native_serde.py, which would also fix the problem

Reply:
cause then you could just call syft.serde._simplify on the `hisotry_cb.history`

Reply:
good point, we don't need this anymore. We are now serializing numpy numbers in the main PySyft https://github.com/dropoutlabs/PySyft/pull/16. Thanks